File: generated-code/codecraft/go/model/attack_properties.go
package model

import "io"
import . "trans_gen_test/stream"
import "fmt"

type AttackProperties struct {
    AttackRange int32
    Damage int32
    CollectResource bool
}

func NewAttackProperties(attackRange int32, damage int32, collectResource bool) AttackProperties {
    return AttackProperties {
        AttackRange: attackRange,
        Damage: damage,
        CollectResource: collectResource,
    }
}

func ReadAttackProperties(reader io.Reader) AttackProperties {
    var attackRange int32
    attackRange = ReadInt32(reader)
    var damage int32
    damage = ReadInt32(reader)
    var collectResource bool
    collectResource = ReadBool(reader)
    return AttackProperties {
        AttackRange: attackRange,
        Damage: damage,
        CollectResource: collectResource,
    }
}

func (attackProperties AttackProperties) Write(writer io.Writer) {
    attackRange := attackProperties.AttackRange
    WriteInt32(writer, attackRange)
    damage := attackProperties.Damage
    WriteInt32(writer, damage)
    collectResource := attackProperties.CollectResource
    WriteBool(writer, collectResource)
}

func (attackProperties AttackProperties) String() string {
    stringResult := "{ "
    stringResult += "AttackRange: "
    attackRange := attackProperties.AttackRange
    stringResult += fmt.Sprint(attackRange)
    stringResult += ", "
    stringResult += "Damage: "
    damage := attackProperties.Damage
    stringResult += fmt.Sprint(damage)
    stringResult += ", "
    stringResult += "CollectResource: "
    collectResource := attackProperties.CollectResource
    stringResult += fmt.Sprint(collectResource)
    stringResult += " }"
    return stringResult
}

File: generated-code/codecraft/csharp/Model/AttackProperties.cs
namespace TransGenTest.Model
{
    public struct AttackProperties
    {
        public int AttackRange { get; set; }
        public int Damage { get; set; }
        public bool CollectResource { get; set; }
    
        public AttackProperties(int attackRange, int damage, bool collectResource)
        {
            this.AttackRange = attackRange;
            this.Damage = damage;
            this.CollectResource = collectResource;
        }
    
        public static AttackProperties ReadFrom(System.IO.BinaryReader reader)
        {
            var result = new AttackProperties();
            result.AttackRange = reader.ReadInt32();
            result.Damage = reader.ReadInt32();
            result.CollectResource = reader.ReadBoolean();
            return result;
        }
    
        public void WriteTo(System.IO.BinaryWriter writer)
        {
            writer.Write(AttackRange);
            writer.Write(Damage);
            writer.Write(CollectResource);
        }
    
        public override string ToString() {
            string stringResult = "AttackProperties { ";
            stringResult += "AttackRange: ";
            stringResult += AttackRange.ToString();
            stringResult += ", ";
            stringResult += "Damage: ";
            stringResult += Damage.ToString();
            stringResult += ", ";
            stringResult += "CollectResource: ";
            stringResult += CollectResource.ToString();
            stringResult += " }";
            return stringResult;
        }
    }
}